plugins/reports/counter/classes/releases/CounterRelease.inc.php

```diff
@@ -0,0 +1,254 @@
+<?php
+
+/**
+ * @file plugins/generic/sushiLite/classes/releases/CounterRelease.inc.php
+ *
+ * Copyright (c) 2014 University of Pittsburgh
+ * Distributed under the GNU GPL v2 or later. For full terms see the file docs/COPYING.
+ *
+ * @class CounterRelease
+ * @ingroup plugins_reports_counter
+ *
+ * @brief A COUNTER report, base class
+ */
+require_once(dirname(dirname(__FILE__)).'/COUNTER/COUNTER.php');
+
+define('COUNTER_EXCEPTION_WARNING', 0);
+define('COUNTER_EXCEPTION_ERROR', 1);
+define('COUNTER_EXCEPTION_PARTIAL_DATA', 4);
+define('COUNTER_EXCEPTION_NO_DATA', 8);
+define('COUNTER_EXCEPTION_BAD_COLUMNS', 16);
+define('COUNTER_EXCEPTION_BAD_FILTERS', 32);
+define('COUNTER_EXCEPTION_BAD_ORDERBY', 64);
+define('COUNTER_EXCEPTION_BAD_RANGE', 128);
+define('COUNTER_EXCEPTION_INTERNAL', 256);
+
+class CounterRelease {
```

Comment:
My point is that the current way you name this class is confusing. Right now the name of this class gives the idea that reports are inheriting from a release class. Conceptually that's confusing, a report is not a more specific release version. 

But if you think of this class implementing a general report one, it makes sense that every report class should have the information about it's release. If that information and details are too much to stay inside the report class, then you can create separated release classes and let the report ones use them (composition instead of inheritance). But I don't think that's the case, maybe you can decide better.

But for me, to fix that, I suggest you to create a more general report class naming, leave the general implementation as it is now, and then create more specific release reports classes (as you already do) but make sure that the name is representing a report, and not a release. That way we can avoid this confusion.